### ISSUE TITLE
Uncaught TypeError: Cannot read property 'isSame' of undefined

### ISSUE THREAD

**Issue by lianee** [tempusdominus/core#11]
This issue was previously reported on [tempusdominus/bootstrap-4](https://github.com/tempusdominus/bootstrap-4/issues/34) for a long time but it probably belongs to core:

Reproduce:
go to [tempusdominus site, usage, no icon](https://tempusdominus.github.io/bootstrap-4/Usage/#no-icon-input-field-only)
click on field (it is filled with today's date)
immediately clear this field (don't click anywhere, ctrl-A delete will do)
click outside

console will display:
`tempusdominus-bootstrap-4.js:1947 Uncaught TypeError: Cannot read property 'isSame' of undefined
    at TempusDominusBootstrap4._updateMonths (tempusdominus-bootstrap-4.js:1947)
    at TempusDominusBootstrap4._fillDate (tempusdominus-bootstrap-4.js:2116)
    at TempusDominusBootstrap4._update (tempusdominus-bootstrap-4.js:401)
    at TempusDominusBootstrap4._setValue (tempusdominus-bootstrap-4.js:430)
    at TempusDominusBootstrap4._change (tempusdominus-bootstrap-4.js:493)
    at Function._jQueryHandleThis (tempusdominus-bootstrap-4.js:2699)
    at r.fn.init._jQueryInterface [as datetimepicker] (tempusdominus-bootstrap-4.js:2706)
    at HTMLInputElement.<anonymous> (tempusdominus-bootstrap-4.js:2734)
    at HTMLDocument.dispatch (jquery-3.2.1.slim.min.js:3)
    at HTMLDocument.q.handle (jquery-3.2.1.slim.min.js:3)
_updateMonths @ tempusdominus-bootstrap-4.js:1947
_fillDate @ tempusdominus-bootstrap-4.js:2116
_update @ tempusdominus-bootstrap-4.js:401
_setValue @ tempusdominus-bootstrap-4.js:430
_change @ tempusdominus-bootstrap-4.js:493
_jQueryHandleThis @ tempusdominus-bootstrap-4.js:2699
_jQueryInterface @ tempusdominus-bootstrap-4.js:2706
(anonymous) @ tempusdominus-bootstrap-4.js:2734
dispatch @ jquery-3.2.1.slim.min.js:3
q.handle @ jquery-3.2.1.slim.min.js:3
trigger @ jquery-3.2.1.slim.min.js:3
(anonymous) @ jquery-3.2.1.slim.min.js:3
each @ jquery-3.2.1.slim.min.js:2
each @ jquery-3.2.1.slim.min.js:2
trigger @ jquery-3.2.1.slim.min.js:3
_notifyEvent @ tempusdominus-bootstrap-4.js:567
_setValue @ tempusdominus-bootstrap-4.js:425
_change @ tempusdominus-bootstrap-4.js:493
_jQueryHandleThis @ tempusdominus-bootstrap-4.js:2699
_jQueryInterface @ tempusdominus-bootstrap-4.js:2706
(anonymous) @ tempusdominus-bootstrap-4.js:2734
dispatch @ jquery-3.2.1.slim.min.js:3
q.handle @ jquery-3.2.1.slim.min.js:3
tempusdominus-bootstrap-4.js:2501 Uncaught TypeError: Cannot read property 'clone' of undefined
    at TempusDominusBootstrap4.hide (tempusdominus-bootstrap-4.js:2501)
    at Function._jQueryHandleThis (tempusdominus-bootstrap-4.js:2699)
    at r.fn.init._jQueryInterface [as datetimepicker] (tempusdominus-bootstrap-4.js:2706)
    at HTMLInputElement.<anonymous> (tempusdominus-bootstrap-4.js:2744)
    at HTMLDocument.dispatch (jquery-3.2.1.slim.min.js:3)
    at HTMLDocument.q.handle (jquery-3.2.1.slim.min.js:3)
    at Object.trigger (jquery-3.2.1.slim.min.js:3)
    at Object.simulate (jquery-3.2.1.slim.min.js:3)
    at HTMLDocument.c (jquery-3.2.1.slim.min.js:3)
hide @ tempusdominus-bootstrap-4.js:2501
_jQueryHandleThis @ tempusdominus-bootstrap-4.js:2699
_jQueryInterface @ tempusdominus-bootstrap-4.js:2706
(anonymous) @ tempusdominus-bootstrap-4.js:2744
dispatch @ jquery-3.2.1.slim.min.js:3
q.handle @ jquery-3.2.1.slim.min.js:3
trigger @ jquery-3.2.1.slim.min.js:3
simulate @ jquery-3.2.1.slim.min.js:3
c @ jquery-3.2.1.slim.min.js:3`

**Comment by nickalbrecht** [comment]
Seems like this is still a bug? Came here from tempusdominus/bootstrap-4#34 as well. Bug is almost 2 years old. I hate pinging authors directly, but maybe you can take a look @Eonasdan and see if if there's a way to correct the library from encountering errors when trying to clear a date?

**Comment by jayaddison** [comment]
It looks like this is somehow related to bootstrap-4`'s call to [this._getLastPickedDate()](https://github.com/producerism/bootstrap-4/commit/b85898ec767e7a2417208ba61adb6705058cb955#diff-aafbc239c2c17bf4a488b9134bd4f651R347).

That call appears to be routed to the global [DateTimePicker](https://github.com/producerism/bootstrap-4/commit/b85898ec767e7a2417208ba61adb6705058cb955#diff-aafbc239c2c17bf4a488b9134bd4f651R2) that is exported by core (in the originally-referenced 5.0.0-alpha6 [here](https://github.com/tempusdominus/core/blob/56b3f082e77e35b16630b40de1c1a58d246e14e4/src/js/tempusdominus-core.js#L1441) and the current HEAD [here](https://github.com/tempusdominus/core/blob/0f5fff683014dba8312e6060db6aff96686c6194/src/js/tempusdominus-core.js#L1513)).

Is it possible that the semantics and access of the `this` keyword have changed somehow -- perhaps as a result of changes in the Gruntfile ([diff](https://github.com/tempusdominus/core/compare/56b3f082e77e35b16630b40de1c1a58d246e14e4..0f5fff683014dba8312e6060db6aff96686c6194#diff-35b4a816e0441e6a375cd925af50752c) - `uglify` and `nugetpack` stand out as changes)?

**Comment by tonix-tuft** [comment]
@Eonasdan 

You can fix it in the following way:

```
            months.removeClass('active');
            // Check if "lastPickedDate" is truthy otherwise .isSame will throw "TypeError: Cannot read property 'isSame' of undefined"
            var lastPickedDate = this._getLastPickedDate();
            if (lastPickedDate && lastPickedDate.isSame(this._viewDate, 'y') && !this.unset) {
                months.eq(this._getLastPickedDate().month()).addClass('active');
            }
```

Will create a PR for this.

**Comment by Eonasdan** [comment]
fixed via #31
